src/components/Api.js

```diff
@@ -0,0 +1,115 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+}
+
+_checkResponse(res) {
+  if (res.ok) {
+    return res.json();
+  }
+  return Promise.reject(`Error: ${res.status}`);
+} 
+  
+  // fetch the info from the database 
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+      })
```

Comment:
`.then(this._checkResponse)` should be used in any request inside `Api.js`, to check if the response is `ok` or not